packages/test_track/lib/src/networking/interceptors/retry_interceptor.dart

```diff
@@ -36,24 +47,59 @@ class RetryInterceptor extends Interceptor {
 
       final extraOptions = extra.toExtraOptions()
         ..addAll(
-          <String, dynamic>{HttpClient.isIdempotentOptionsKey: isIdempotent},
+          <String, dynamic>{isIdempotentOptionsKey: isIdempotent},
         );
 
       try {
-        return await _dio.request<dynamic>(
+        final retry = RawRequest(
           err.requestOptions.path,
+          type: requestType,
           cancelToken: err.requestOptions.cancelToken,
-          data: err.requestOptions.data,
+          data: NetworkRequestBody.raw(err.requestOptions.data),
           onReceiveProgress: err.requestOptions.onReceiveProgress,
           onSendProgress: err.requestOptions.onSendProgress,
           queryParameters: err.requestOptions.queryParameters,
           options: Options(extra: extraOptions),
         );
-      } on DioError catch (err) {
+        await _clientGetter().execute<dynamic, void>(
+          retry,
+          onResponse: (r) {
+            DioException errorForRequest() {
+              return DioException(requestOptions: err.requestOptions);
+            }
+
+            // We don't have access to the raw Response from `sturdy_http`, so
+            // for both success cases, fabricate them. This should be fine for
+            // most cases, and would require a `sturdy_http` change if we want
+            // to gain access to the raw response.
+            r.when(
+              ok: (res) => handler.resolve(
+                Response(
+                  data: res,
+                  requestOptions: err.requestOptions,
+                ),
+              ),
+              okNoContent: () => handler.resolve(
+                Response(
+                  statusCode: 204,
+                  requestOptions: err.requestOptions,
+                ),
+              ),
+              genericError: (_, __, error) => throw error ?? errorForRequest(),
```

Comment:
in what scenario would we actually hit `errorForRequest`? can `error` ever be null? i know the `genericError` is the default case for when the error code falls through.